### ISSUE TITLE
possible memory leak in ASIHTTPRequest.m

### ISSUE THREAD

**Issue by ghost** [pokeb/asi-http-request#269]
in method (void)startRequest of ASIHTTPRequest class, at line 7, sslProperties is allocated but not released (nor autorelease). Xcode analyze warns that it'll cause memory leak.

```
  if([[[[self url] scheme] lowercaseString] isEqualToString:@"https"]) {       

    // Tell CFNetwork not to validate SSL certificates
    if (![self validatesSecureCertificate]) {
      // see: http://iphonedevelopment.blogspot.com/2010/05/nsstream-tcp-and-ssl.html

      NSDictionary *sslProperties = [[NSDictionary alloc] initWithObjectsAndKeys:
                                     [NSNumber numberWithBool:YES], kCFStreamSSLAllowsExpiredCertificates,
                                     [NSNumber numberWithBool:YES], kCFStreamSSLAllowsAnyRoot,
                                     [NSNumber numberWithBool:NO],  kCFStreamSSLValidatesCertificateChain,
                                     kCFNull,kCFStreamSSLPeerName,
                                     nil];

      CFReadStreamSetProperty((CFReadStreamRef)[self readStream], 
                              kCFStreamPropertySSLSettings, 
                              (CFTypeRef)sslProperties);
    } 
```

**Comment by bmauter** [comment]
Just found this via Google search because my Analyzer also kicked that message out.  I tried to read through the Apple docs to see if `CFReadStreamSetProperty` increments the retain count on the third parameter, but I'm not finding anything.  I put a pair of `NSLog(@"retain count=%d", [sslProperties retainCount]);` statements around the `CFReadStreamSetProperty` call and I could see the retain count was in fact incremented at runtime.

So, I changed the alloc and init line to `[NSDictionary dictionaryWithObjectsAndKeys:` and I'm not seeing an issue yet.  This does in fact appear to be a real leak and not just a potential one.  You'll only encounter it if you use https and you've instructed ASIHTTPRequest to not validate SSL certificates.
